Describe how to set up Athena with S3 data (#596)
objectiv/objectiv.io: docs/docs/tracking/collector/s3.md

To check if the messages have successfully arrived in the queue, please review to monitoring in the AWS 
console. Events should show up as either `PutRecords` (Kinesis) or `Number of messages received (SQS).

> 

Author: I have the feeling that there are some steps missing above. It seems the steps above get the data in SQS, but not quite to S3. That's @borft 's expertise tho.

Reviewer: The steps that are missing involve setting up a working Snowplow pipeline. I'd say, at least at the time, that was considered beyond the scope of this document.

FYI: what's happening is:
objectiv-collector -> MQ -> enrich-kinesis -> MQ -> s3-loader -> s3

The part beyond the Objectiv Collector is considered to be part of the Snowplow pipeline

Author: After our chat I added a little bit more info in 5c3e3de834cb1b5e3fc3e961c970607180cd7f8c

> ## Accessing data with Athena

Reviewer 2: ```suggestion
## Use Objectiv with Athena
```

> To be able to use the data on S3 with Bach and modelhub, a few steps need to be taken:

Reviewer 2: ```suggestion
To be able to use the data on S3 with [Bach](/modeling/bach/index.mdx) and [the open model hub](/modeling/open-model-hub/index.mdx), a few steps need to be taken:
```

>  1. An Athena table must be created that makes the S3 data queryable.
>  2. Additional Athena setup: permissions and query-result storage
>  3. Constructing the URL to connect with Athena.

Reviewer 2: ```suggestion
 1. Create an Athena table that makes the S3 data queryable.
 2. Perform additional Athena setup: permissions and query-result storage.
 3. Construct the URL to connect with Athena.
```

    mkt_clickid string,
    mkt_network string,
    etl_tags string,
    dvce_sent_tstamp timestamp,
    refr_domain_userid string,
    refr_device_tstamp timestamp,
    derived_contexts string,
    domain_sessionid string,
    derived_tstamp timestamp,
    event_vendor string,
    event_name string,
    event_format string,
    event_version string,
    event_fingerprint string,
    true_tstamp timestamp
)
ROW FORMAT DELIMITED
FIELDS TERMINATED BY '\t'
ESCAPED BY '\\'
LINES TERMINATED BY '\n'
LOCATION 's3://<bucket>/<prefix>>/'   -- Example: LOCATION 's3://objectiv-production-data/enriched/';
;

-- Column names and types were inspired by Snowplow's create_glue_table.sh
```

### Additional Athena setup
All Athena query results are written to S3, to the so called 'S3 staging'. Therefore, an s3 bucket or prefix
should be created that can be used for that purpose. This S3 prefix should not overlap with the location of
the data.

An account should be setup, with the following access:
* S3:
  * read access to the data, i.e. the `LOCATION` that the `events` table gets its data from.
  * write access to the S3 staging prefix.
* Athena: access to query/create/etc. tables.

> * Glue: access to Glue data catalog.

Author: I want to do this a bit more in detail, similar to what was done here: https://github.com/objectiv/objectiv-analytics/blob/main/bach/tests/athena-setup-test-db.md. But that requires some more research. I think the information we have here now is sufficient for now.

### Construct Athena URL
To use the Objectiv modelhub with Athena, in Python one must simply provide a url to connect to the database.
The general form of a URL for Athena is:

```
awsathena+rest://{aws_access_key_id}:{aws_secret_access_key}@athena.{region_name}.amazonaws.com:443/
{schema_name}?s3_staging_dir={s3_staging_dir}
```

With the following fields set:
* `aws_access_key_id`: account key id.
* `aws_secret_access_key`: secret access key
* `region_name`: region, e.g. `eu-west-1`
* `schema_name`: name of database that contains the `events` table defined earlier
* `s3_staging_dir`: path to S3 where query results can be stored

Additionally, there are optional query string parameters that can be added to the url:
* `work_group`: work-group under which the queries run
* `catalog_name`: catalog that contains the table defintions

Note that all values must be properly escaped. As a help, the modelhub offers a function that constructs the
url: `athena_construct_engine_url()`.

```python

> from modelhub import athena_construct_engine_url

Author: This function will soon be added: https://github.com/objectiv/objectiv-analytics/pull/1416

* S3:
  * read access to the data, i.e. the `LOCATION` that the `events` table gets its data from.
  * write access to the S3 staging prefix.
* Athena: access to query/create/etc. tables.
* Glue: access to Glue data catalog.

### Construct Athena URL
To use the Objectiv modelhub with Athena, in Python one must simply provide a url to connect to the database.
The general form of a URL for Athena is:

```
awsathena+rest://{aws_access_key_id}:{aws_secret_access_key}@athena.{region_name}.amazonaws.com:443/
{schema_name}?s3_staging_dir={s3_staging_dir}
```

With the following fields set:
* `aws_access_key_id`: account key id - optional
* `aws_secret_access_key`: secret access key - optional
* `region_name`: region, e.g. `eu-west-1`
* `schema_name`: name of database that contains the `events` table defined earlier
* `s3_staging_dir`: path to S3 where query results can be stored

Additionally, there are optional query string parameters that can be added to the url:
* `work_group`: work-group under which the queries run
* `catalog_name`: catalog that contains the table definitions

If the access key fields are not set, then the Athena library will use default mechanisms to get

> credentials (environment variables, config files, etc. See [Amazon docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#configuring-credentials))

Author: Added here some info on not providing the credentials @borft
